tests/Doctrine/Tests/Models/Issue5887/Issue5887CustomIdObjectType.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Doctrine\Tests\Models\Issue5887;
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Types\Type;
+
+class Issue5887CustomIdObjectType extends Type
```

Comment:
Please move this class to the test file (like we have [here](https://github.com/doctrine/doctrine2/blob/e4704beaf9fad5a521fe7592dec382ae209b3cc1/tests/Doctrine/Tests/ORM/Functional/Ticket/GH6141Test.php)).